### PR TITLE
Feature: macOS, watchOS, tvOS support

### PR DESCRIPTION
**Pull request by Brett-Best**

### CODE REVIEW

File: Sources/PureSwiftUI/Extensions/Convenience/CoreGraphics/CGRect+Convenience.swift [hunk 1]
```diff
@@ -183,9 +183,13 @@ public extension CGRect {
...
-    
+
     func inset(_ top: CGFloat, _ leading: CGFloat, _ bottom: CGFloat, _ trailing: CGFloat) -> CGRect {
-        inset(by: UIEdgeInsets(top: top, left: leading, bottom: bottom, right: trailing))
+      #if os(macOS)
+      return self
+      #else
+      return inset(by: UIEdgeInsets(top: top, left: leading, bottom: bottom, right: trailing))
+      #endif
     }
```
Comment (Brett-Best) [inline Sources/PureSwiftUI/Extensions/Convenience/CoreGraphics/CGRect+Convenience.swift:252]: I wasn’t sure what to replace this with on macOS, would be great if you could take a look!

File: Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift [hunk 2]
```diff
@@ -14,7 +14,11 @@ public struct DynamicallyScalingFontViewModifier: ViewModifier {
     var weight: Font.Weight?
 
     public func body(content: Content) -> some View {
+        #if os(macOS)
+        let scaledSize = CGFloat(17)
+        #else
         let scaledSize = UIFontMetrics.default.scaledValue(for: size)
+        #endif
         return content.font(createFont(scaledSize: scaledSize))
     }
```
Comment (Brett-Best) [inline Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift:23]: I wasn’t sure what to replace this with either on macOS, I wonder what the macCatalyst implementation calls through to in AppKit. Would be great if you could take a look, perhaps having a hardcoded value is okay.

**Review by Brett-Best — COMMENTED** [review]
@CodeSlicing would appreciate if you could have a look at this!
I could also make a PR for making sure the project builds (+ maybe run the unit tests), see https://github.com/SwiftUIX/SwiftUIX/runs/1471421259 for an example of what I’ve setup (:

**Comment by CodeSlicing** [comment]
Hey there,

Sorry for the lack of progress on this! I have so little time for anything at the moment that it's difficult to carve out the time to do a proper review.

V interested in the automated tests on commit.

Promise to get back to you on this as soon as I can dedicate the time to it. 

All the best,

Adam

**Comment by CodeSlicing** [comment]
Once I've got everything merged back to develop and looking good as it stands, I'll make a move on this your PR :)

**Comment by Brett-Best** [comment]
@CodeSlicing be great if you could take a look at this before WWDC come around, was a bit of a pain to merge your latest changes in

**Comment by CodeSlicing** [comment]
Sorry about that - had to merge everything to develop to get things back on track. 

As it stands the plan is to separate the drawing and many of the lower-level convenience extensions into a separate package that will then be imported by PureSwiftUI. It's something people have been asking for and makes sense to me. 

With that in mind, I've decided it's version 3 of PureSwiftUI that's going to get your changes to match up with the next version of SwiftUI. I'll be taking time off work to specifically look at things like this. 

I realise it's frustrating. This is why in the readme I was quite explicit in not open-sourcing this project as I have so little time to administer PRs and I don't want to mess people about. However, I do have a plan to merge the changes you've been kind enough to work on :)

**Comment by Brett-Best** [comment]
No need to apologise 😊, that sounds great to me. Fingers crossed Xcode 13 won’t require many changes to get things compiling!

File: Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift (hunk 2 above)
Comment (OskarGroth) [reply Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift:23]: Dynamic type does not exist on macOS. The docs for the `sizeCategory` environment value (now renamed `dynamicTypeSize`) states that:

> On macOS, this value cannot be changed by users and does not affect the text size

Just make this whole modifier unavailable on macOS.

**Comment by OskarGroth** [comment]
@CodeSlicing I'd love to see macOS support for this lib. Any reason why this PR can't be merged as-is? I get that you're planning for major refactors but it seems to be pretty minor changes and life tends to get in the way (we're soon 1 year down from this PR).

**Comment by CodeSlicing** [comment]
Yes indeed, I'm currently working on 3.0.0 of PureSwiftUI which will incorporate this PR! The idea is to release 3.0.0 in September following the official release of iOS15 etc.

File: Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift (hunk 2 above)
Comment (CodeSlicing) [reply Sources/PureSwiftUI/ViewModifiers/Font/DynamicallyScaledFontViewModifier.swift:23]: yes completely agree. Once this PR is applied, I'll move the conditional compilation to include the entire extension.

**Comment by CodeSlicing** [comment]
Ok merging this into feature/2.3.0 and will cherry pick into feature/3.0.0 of PureSwiftUI and feature/1.0.0 of PureSwiftUIDesign.

**Comment by CodeSlicing** [comment]
PR is now part of tag '2.3.0-beta-2' - will update this PR when appropriate changes have been cherry picked into the other versions I mentioned.

**Comment by CodeSlicing** [comment]
scratch that - it's now '2.3.0-beta-4' - had to make some changes in the conditional compilation for watchOS

**Comment by CodeSlicing** [comment]
All very exciting stuff! Thanks Brett, and sorry it took sooooooo long ;o)

**Pull request merged**